### PR TITLE
Created new workflow for nightly ci

### LINKED ISSUE #1347: run nightly regression test
Currently we have the [regression workflow](https://github.com/buildtesters/buildtest/blob/devel/.github/workflows/regression.yml) run on every PR which is okay for development. 

Let's extend the CI check to run nightly regression test on the latest release which is the `master` branch. 

See https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#schedule for some tips on how to run scheduled workflows 

@prathmesh4321 this could be a good task for you to implement if you are new to github workflows.

### PR DESCRIPTION
**Pull request by prathmesh4321**
Hi @shahzebsiddiqui . I have the new workflow for nightly regression. Please review.

### CODE REVIEW

**Comment by shahzebsiddiqui** [comment]
please add the ci badge to `README` file and then we can try to clean up the `on` event to just limit to `schedule`. We should let it run every 5min and see if it works. If it doesn't we may need to have

```
on:
  pull_requests:
    branches: [devel]
  push:
    branches: [devel]
```

File: .github/workflows/nightly_regression.yml [hunk 1]
```diff
@@ -0,0 +1,53 @@
+name: Nightly Regression Test
+
+on:
+  push:
+    branches: [devel]
+  pull_request:
+    branches: [devel]
```
Comment (shahzebsiddiqui) [inline .github/workflows/nightly_regression.yml]: lets remove this and then we can see how this goes tomorrow

File: .github/workflows/nightly_regression.yml [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+  schedule:
+    - cron: '*/5 * * * *'
```
Comment (shahzebsiddiqui) [inline .github/workflows/nightly_regression.yml]: change this back to once a day.

Comment (prathmesh4321) [reply .github/workflows/nightly_regression.yml]: @shahzebsiddiqui Okay

**Pull request merged**